numpy/f2py/backends/meson_backend.py

```diff
@@ -0,0 +1,183 @@
+		self.module_name = module_name
+		self.numpy_install_path = numpy_install_path
+		self.build_template_path = numpy_install_path / "f2py" / "backends" / "src" / "meson.build.src"
+		self.sources = fortran_sources
+		self.numpy_get_include = numpy_get_include
+		self.f2py_get_include = f2py_get_include
+		self.wrappers = wrappers
+		self.dependencies = dependencies
+		self.include_directories = include_path
+		self.substitutions = {}
+		self.optimization_flags = optimization_flags
+		self.architecture_flags = architecture_flags
+		self.fortran_flags = f77_flags + f90_flags
+		self.linker_libpath = linker_libpath
+		self.linker_libname = linker_libname
+		self.define_macros = define_macros
+		self.undef_macros = undef_macros
+		self.pipeline = [self.initialize_template,
+						self.global_flags_substitution,
+                   		self.sources_substitution,
+                      	self.dependencies_substitution,
+                       	self.include_directories_subtitution,
+                        self.linker_substitution,
+                        self.macros_substitution]
+
+	@property
+	def meson_build_template(self) -> str:
+		if(not self.build_template_path.is_file()):
+			raise FileNotFoundError(errno.ENOENT, f"Meson build template {self.build_template_path.absolute()} does not exist.")
+		return self.build_template_path.read_text()
+
+	def initialize_template(self) -> None:
+		"""Initialize with module name and external NumPy and F2PY C libraries."""
+		self.substitutions['modulename'] = self.module_name
+		self.substitutions['numpy_get_include'] = self.numpy_get_include.absolute()
+		self.substitutions['f2py_get_include'] = self.f2py_get_include.absolute()
+	
+	def sources_substitution(self) -> None:
+		self.substitutions["source_list"] = ",".join(["\'"+str(source.absolute())+"\'" for source in self.sources])
+		self.substitutions["wrappers"] = ",".join(["\'"+str(wrapper.absolute())+"\'" for wrapper in self.wrappers])
+
+	def dependencies_substitution(self) -> None:
+		self.substitutions["dependencies_list"] = ", ".join([f"dependency('{dependecy}')" for dependecy in self.dependencies])
+		
+	def include_directories_subtitution(self) -> None:
+		self.substitutions["include_directories_list"] = ", ".join([f"include_directories('{include_directory}')" for include_directory in self.include_directories])
+
+	def global_flags_substitution(self) -> None:
+		fortran_compiler_flags = self.fortran_flags + self.optimization_flags + self.architecture_flags
+		c_compiler_flags = self.optimization_flags + self.architecture_flags
+		self.substitutions["fortran_global_args"] = fortran_compiler_flags
+		self.substitutions["c_global_args"] = c_compiler_flags
```

Comment:
Global flags are generally a bad idea, especially since a lot of these might be used as subprojects, it makes more sense to populate the `_project` variants.